Asset Observations Doc

clairelin135 opened the pull request.

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,99 @@
+def observation_asset():
+    df = read_df()
+    remote_storage_path = persist_to_storage(df)
+    yield AssetObservation(asset_key=AssetKey("observation_asset"), metadata={"num_rows": len(df)})
```
This reminds me that we should add the ability to do `context.log_asset_observation` in the same way that users can now do `context.log_asset_materialization`.

clairelin135 replied:
It's already been added [here](https://github.com/dagster-io/dagster/blob/25abc874d13decf8efdc01899f112aacced8d3e3/python_modules/dagster/dagster/core/execution/context/compute.py#L300)

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,99 @@
+using the <PyObject object="asset" decorator /> decorator or have existing materializations
+in order for its observations to be displayed.
+
+## Yielding an AssetObservation from an Asset
```
Because ops / jobs are the APIs that we forefront on the rest of the site, I think we should have the primary examples here be in terms of ops, instead of software-defined assets.

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,99 @@
+| Name                                                    | Description                                                          |
+| ------------------------------------------------------- | -------------------------------------------------------------------- |
+| <PyObject module="dagster" object="AssetObservation" /> | Dagster event indicating that an asset's metadata has been recorded. |
+| <PyObject module="dagster" object="AssetKey" />         | A unique identifier for a particular external asset                  |
```
Nitpick: add period at end of sentence

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,99 @@
+
+## Overview
+
+<PyObject module="dagster" object="AssetObservation" /> events are used to communicate
```
I think the first sentence might be clearer as "are used to record metadata in Dagster about a given asset".

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,99 @@
+
+<PyObject module="dagster" object="AssetObservation" /> events are used to communicate
+to Dagster that metadata has been recorded about a given asset. Asset observation
+events can be yielded at runtime within ops and assets. An asset must be defined
```
This reminds me that we should make it possible to log AssetObservations from `load_input` functions of IO managers.

clairelin135 replied:
Adding this

clairelin135 commented:
@sryza I addressed the feedback above and added a snippet to the op events page.

All of the op examples also yield asset materializations, because if users execute the code the asset will not display in the asset catalog without a materialization. I plan on updating this doc to contain IO manager examples if #6653 can be merged in time.

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,105 @@
+
+### Attaching Metadata to an AssetObservation
+
+There are a variety of types of metadata that can be associated with an observation event, all through the <PyObject object="EventMetadataEntry" /> class. Each observation event optionally takes a list of metadata entries that are then displayed in the event log and the [Asset Details](/concepts/dagit/dagit#asset-details) page. Check our API docs for <PyObject object="EventMetadataEntry" /> for m ...
```
list -> dictionary

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,105 @@
+            "size (bytes)": calculate_bytes(df),
+        },
+    )
+    yield AssetMaterialization(asset_key="my_dataset")
```
Same comment as above: probably better to leave out the AssetMaterialization

clairelin135 replied:
I can leave out the AssetMaterialization for the examples that don't link to the asset details page. This example links to asset details page and without the asset materialization, users may not be able to nav to the asset details page so I think we should keep it in

sryza commented on docs/content/concepts/assets/asset-observations.mdx:
```diff
@@ -0,0 +1,105 @@
+We should now see an observation event in the event log when we execute this asset.
+
+<Image
+alt="asset-observation"
```
Are we able to add an entry to screenshots.yaml for generating this screenshot?

clairelin135 replied:
added

sryza reviewed (approved):
Looks great aside from a couple comments
